File: packages/openmemory-py/src/openmemory/server/routers/compression.py
from fastapi import APIRouter, HTTPException, Depends, Body
from typing import Dict, Any, List, Optional
from pydantic import BaseModel
from ...ops.compress import compression_engine
from ...core.types import CompressionResult, CompressionStats

router = APIRouter(tags=["compression"])

class CompressRequest(BaseModel):
    text: str
    algorithm: str = "semantic"

@router.post("/test")
async def test_compression(req: CompressRequest):
    """
    Test compression algorithm on input text.
    """
    try:
        if req.algorithm == "auto":
            res = compression_engine.auto(req.text)
        else:
            res = compression_engine.compress(req.text, req.algorithm)
        
        return {"success": True, "result": res}
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))

@router.get("/stats")
async def get_compression_stats():
    """
    Get compression engine statistics.
    """
    stats = compression_engine.get_stats()
    return {"success": True, "stats": stats}

File: packages/openmemory-js/src/clients/compression.ts
/**
 * @file compression.ts
 * @description sub-client for Text Compression operations.
 * @audited 2026-01-19
 */

import { BaseSubClient } from "./base";
import type {
    CompressionResult,
    CompressionStats,
} from "../core/types";

/**
 * Text Compression Operations sub-client.
 */
export class CompressionClient extends BaseSubClient {
    /**
     * Compress a text string using various algorithms.
     */
    async compress(
        text: string,
        algorithm: "semantic" | "syntactic" | "aggressive" = "semantic",
        userId?: string,
        options?: { signal?: AbortSignal }
    ): Promise<CompressionResult> {
        const uid = userId || this.defaultUser;
        const res = await this.request<{ result: CompressionResult }>("/api/compression/test", {
            method: "POST",
            body: JSON.stringify({ text, algorithm, userId: uid }),
            signal: options?.signal,
        });
        return res.result;
    }

    /**
     * Get compression engine statistics.
     */
    async getStats(options?: { signal?: AbortSignal }): Promise<CompressionStats> {
        const res = await this.request<{ stats: CompressionStats }>("/api/compression/stats", { signal: options?.signal });
        return res.stats;
    }
}
